src/v/raft/consensus.cc

```diff
@@ -277,6 +277,11 @@ ss::future<> consensus::stop() {
         co_await _snapshot_writer->close();
         _snapshot_writer.reset();
     }
+    /**
+     * Clear metrics after consensus instance is stopped.
+     */
+    _metrics.clear();
+    _probe->clear();
```

Comment:
nitpick:
```suggestion
    _probe.reset();
```
I wonder if we should audit https://github.com/redpanda-data/redpanda/pull/11160 and stop all probes when a service stops.

Reply:
i wanted to make this more explicit. The same pattern is used in `storage::probe`